collab2/foraging/toolkit/filtering.py

```diff
@@ -0,0 +1,35 @@
+from typing import Callable, List, Optional
+
+import numpy as np
+import pandas as pd
+
+
+# PP_comment : even with passing constraint params as a dictionary, we might have a limitation :
```

Comment:
I think the proper workaround is to instantiate object, add velocities at then use the object to write a constraint because now it will have velocities available and then pass this constraint to predictor derivation. If we can isolate velocities addition from derived predictor calculation - which is the way I had it, and I think it's a pretty natural way to go. Something to keep in mind, and something definitely worth trying at a point when we're working on velocity.

Reply:
Ah okay. so maybe the add_velocity function could be called inside the wrapper (non-underscored function) which updates the foragersDF and then this df is passed to the underscored function (where it is used in the constraint)

Reply:
yeah, I think that's one way to do it